features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/draft/VolatileComposerDraftStore.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2024 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.features.messages.impl.draft
+
+import io.element.android.libraries.matrix.api.core.RoomId
+import io.element.android.libraries.matrix.api.room.draft.ComposerDraft
+import javax.inject.Inject
+
+/**
+ * A volatile draft store that keeps drafts in memory only.
+ * It can be used to store drafts that should not be persisted across app restarts.
+ * Currently it's used to store draft message when moving to edit mode.
+ */
+class VolatileComposerDraftStore @Inject constructor() : ComposerDraftStore {
+    private val drafts: MutableMap<RoomId, ComposerDraft> = mutableMapOf()
```

Comment:
It's weird to store a map since a new `VolatileComposerDraftStore` will be created for each room I think.